cmd/artifact/main.go

```diff
@@ -0,0 +1,341 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	tcclient "github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/queue"
+	artifact "github.com/taskcluster/taskcluster-lib-artifact-go"
+	"github.com/urfave/cli"
+)
+
+const (
+	ErrInternal = 1
+	ErrBadUsage = 2
+	ErrCorrupt  = 13
+)
+
+// TODO implement an in memory 'file'
+// TODO implement 'redirect' and 'error' artifact types?
+
+func main() {
+	err := _main(os.Args)
+	if err == nil {
+		os.Exit(0)
+	}
+
+	if ecErr, ok := err.(cli.ExitCoder); ok {
+		os.Exit(ecErr.ExitCode())
+	}
+
+	os.Exit(ErrInternal)
+
+}
+
+func _main(args []string) error {
+	// We're going to take care of exiting ourselves
+	cli.OsExiter = func(c int) {}
```

Comment:
Too bad you don't like `docopt` 😜

Reply:
I originally tried it, actually.  I found the lack of "real" subcommand support off-putting, but the good news is that this library tackles the drift between docs and reality by generating really nice help messages.

Reply:
fwiw now i've seen this library, i like it a lot too - it is quite intuitive :-)